citations-impl/impl/src/java/org/sakaiproject/citation/impl/DbCitationService.java

```diff
@@ -569,6 +573,9 @@ else if (currentH1Node!=null){
 						currentH1Node.addChild(citationCollectionOrder);
 					}
 				}
+				else if (citationCollectionOrder.getSectiontype().equals(CitationCollectionOrder.SectionType.DESCRIPTION)){
+					lastNode.addChild(citationCollectionOrder);
```

Comment:
Yes you're right.  Bad data like this was the general reason one of the temps got a stackoverflow when editing a reading list, because she (somehow) dragged a h3 after an h1.   There's a Jira to do server-side validation to make sure valid data always gets saved. https://jira.oucs.ox.ac.uk/jira/browse/WL-3935